File: HackerEarth/Magic Pencil.py
# Magic Pencil

# https://www.hackerearth.com/problem/algorithm/problem-1-39/


n=int(raw_input())
for i in range(1,n+1):
	lt=map(int,raw_input().split())
	d=int(pow(lt[0],lt[1]))
	e=int((d/lt[2])*lt[2])
	f=int((d/lt[2]+1)*lt[2])
	g=int(d-e)
	h=int(f-d)
	if g<h:
		print e
	else:
		print f

File: HackerEarth/Magic Pencil.java
// Magic Pencil

// https://www.hackerearth.com/problem/algorithm/problem-1-39/


import java.util.*;
class TestClass {
    public static void main(String args[] ) throws Exception {
    
Scanner s=new Scanner(System.in);
int t=s.nextInt();
while(t!=0)
{
    int a=s.nextInt();  
    int b=s.nextInt();
	int c=s.nextInt();
	int d=(int)Math.pow(a,b); 
	int sum=0,ss=0;
 
     sum=(d/c)*c;
	  ss=((d/c)+1)*c;
	  if(ss-d>=d-sum)
        System.out.println(sum);
        else
        System.out.println(ss);
	
        t--;
}
    }
}